projects/novo-elements/src/elements/data-table/data-table.component.ts

```diff
@@ -66,13 +66,16 @@ import { DataTableState } from './state/data-table-state.service';
       <novo-data-table-pagination
         *ngIf="paginationOptions"
         [theme]="paginationOptions.theme"
-        [length]="dataSource?.currentTotal"
+        [length]="null !== overrideTotal && overrideTotal >= 0 ? overrideTotal : dataSource?.currentTotal"
```

Comment:
should remove `&& overrideTotal >= 0` because zero is a valid value. Your query could return no records.

Reply:
changed to `&& overrideTotal !== undefined`